sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,384 @@
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.nodes
+        [('A', 0, 0)]
+        >>> t.add_node('B', 3, 0)
+        >>> t.nodes
+        [('A', 0, 0), ('B', 3, 0)]
+        """
+        x = sympify(x)
+        y = sympify(y)
+
+        if label in self._node_labels:
+            raise ValueError("Node needs to have a unique label")
+
+        elif x in self._node_position_x and y in self._node_position_y:
+            raise ValueError("A node already exists at the given position")
+
+        else :
+            self._nodes.append((label, x, y))
+            self._node_labels.append(label)
+            self._node_positions.append((x, y))
+            self._node_position_x.append(x)
+            self._node_position_y.append(y)
+            self._loads[label] = [[0, 90]]
+            self._supports[label] = "none"
+
+    def remove_node(self, label):
+        """
+        This method removes a node from the truss.
+
+        Parameters
+        ==========
+        label:  String or Symbol
+            The label of the node to be removed.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.nodes
+        [('A', 0, 0)]
+        >>> t.add_node('B', 3, 0)
+        >>> t.nodes
+        [('A', 0, 0), ('B', 3, 0)]
+        >>> t.remove_node('A')
+        >>> t.nodes
+        [('B', 3, 0)]
+        """
+        for i in range(len(self.nodes)):
+            if self._node_labels[i] == label:
+                x = self._node_position_x[i]
+                y = self._node_position_y[i]
+
+        if label not in self._node_labels:
+            raise ValueError("No such node exists in the truss")
+
+        else :
+            self._nodes.remove((label, x, y))
+            self._node_labels.remove(label)
+            self._node_positions.remove((x, y))
+            self._node_position_x.remove(x)
+            self._node_position_y.remove(y)
+            self._loads.pop(label)
+            self._supports.pop(label)
+
+    def add_member(self, label, start, end):
+        """
+        This method adds a member between any two nodes in the given truss.
+
+        Parameters
+        ==========
+        label: String or Symbol
+            The label for a member. It is the only way to identify a particular member.
+
+        start: String or Symbol
+            The label of the starting point/node of the member.
+
+        end: String or Symbol
+            The label of the ending point/node of the member.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.add_node('C', 2, 2)
+        >>> t.add_member('AB', 'A', 'B')
+        >>> t.members
+        [('AB', 'A', 'B')]
+        """
+
```

Comment:
Throw an error when a member already exists and another member with opposite start and end points is added.
Throw an error when a member is added having same start and end points.